src/_articles/libraries/dart-io.md

```diff
@@ -0,0 +1,386 @@
+---
+title: "An Introduction to the dart:io Library"
+description: "An introduction to the Dart I/O library, which is aimed at code that runs in Flutter and the standalone Dart VM."
+original-date: 2012-03-01
+date: 2018-08-29
+category: libraries
+---
+
+_Written by Mads Ager<br>
+March 2012 (updated August 2018)_
+
+The [dart:io]({{site.dart_api}}/{{site.data.pkg-vers.SDK.channel}}/dart-io/dart-io-library.html) library
+is aimed at code that runs in Flutter and the standalone Dart VM.
+In this article we will give you a feel for
+what is currently possible with dart:io
+by going through a couple of examples.
+
+Dart is a single-threaded programming language.
+If an operation blocks the Dart thread,
+the application makes no progress before that operation completes.
+For scalability it is therefore crucial that no I/O operations block.
+Instead of blocking on I/O operations,
+dart:io uses an asynchronous programming model inspired by
+[node.js,](http://nodejs.org)
+[EventMachine,](https://github.com/eventmachine/eventmachine/wiki) and
+[Twisted.](http://twistedmatrix.com/trac/)
+
```

Comment:
Thinking about this again, maybe we should de-emphasize it a little bit that most of these can be used in Flutter as well, because I think they have more layers on top of the normal `dart:io` and some things should probably not be used (subprocesses).

We could add a section here saying:

## Usage in flutter

Flutter has specific APIs to access assets of an application, which are more high-level than dart:io. Please refer to https://flutter.io/assets-and-images/ for more information.